### PR TITLE
feat: Migrate to RxJava 3 and RxBinding 4 fixes #317

### LINKED ISSUE #317: RxPermissions support for RxJava3
Do you have any plan to update RxPermissions for RxJava3?

### PR DESCRIPTION
**Pull request by galuszkak**

### CODE REVIEW

**Comment by galuszkak** [comment]
Hi @tbruyelle and @epool ,

I've prepared here migration to RxJava 3 and move sample to use RxBinding 4 that also uses RxJava 3.

Tests are passing so I think this PR should be OK. 

Thanks!

**Comment by tbruyelle** [comment]
Looks cleaner than #320 which embeds more changes. Good to me, WDYT @epool ?

File: lib/src/test/java/com/tbruyelle/rxpermissions3/RxPermissionsTest.java [hunk 1]
```diff
@@ -513,7 +488,6 @@ public void eachSubscriptionCombined_trigger_granted() {
         mRxPermissions.onRequestPermissionsResult(new String[]{permission}, result);
 
         sub.assertNoErrors();
-        sub.assertNotTerminated();
```
Comment (epool) [inline lib/src/test/java/com/tbruyelle/rxpermissions3/RxPermissionsTest.java:516]: why these deletions?

Comment (galuszkak) [reply lib/src/test/java/com/tbruyelle/rxpermissions3/RxPermissionsTest.java:516]: Hi @epool ,

Those methods were removed in RxJava 3 from TestObservable. See here:
https://github.com/ReactiveX/RxJava/wiki/What's-different-in-3.0#test-support-methods

Comment (epool) [reply lib/src/test/java/com/tbruyelle/rxpermissions3/RxPermissionsTest.java:516]: got it, thanks!

**Comment by galuszkak** [comment]
@epool answering your question on compatibility with RxJava 2 if anyone would need their Observable to be V2 not V3 they can use bridge library called RxJavaBridge which is available here:
https://github.com/akarnokd/RxJavaBridge

**Comment by tbruyelle** [comment]
Thx @epool for the review
Thx @galuszkak for the clear PR !

**Pull request merged**